concent_api/core/payments/bankster.py

```diff
@@ -362,9 +362,7 @@ def settle_overdue_acceptances(
         )
 
         # Concent defines time T1 equal to youngest timestamp from list of transactions.
-        if len(list_of_transactions) == 0:
-            t1_is_bigger_than_payments_ts = False
-        else:
+        if len(list_of_transactions) > 0:
             youngest_transaction_timestamp = max(
                 ethereum_transaction.closure_time for ethereum_transaction in list_of_transactions
             )
```

Comment:
What if there are no transactions?

Reply:
Then we will never reach this point because view checks that.

Reply:
> Then we will never reach this point

So this means that if the requestor has never paid anything (the list of transactions is empty) bankster will not pay anything either? That's wrong. The whole concept of Concent is that it can pay if requestor does not.

> because view checks that.

That's wrong too. Everything that touches the blockchain is supposed to be inside Bankster. Bankster is an API that wraps SCI and Concent should never communicate with SCI directly now that it's been introduced.

Reply:
Sorry I was wrong, I thought we are talking about client message here. So if there are no transactions there should be an exception raised as it is in current logic, which I have now added. If this is wrong please describe how it should work then.

Reply:
Here's what Lukasz's spec says:

> Usługa Concent tworzy listę transakcji LT składającą się z listy transakcji płatności wykonanych przez Requestora, dla których podany w nich stempel czasu jest nie starszy niż T0. Lista może być pusta. (patrz punkt 4).
> Usługa Concent określa czas T1 równy najmłodszemu stemplowi czasu podanemu w transakcjach z listy LT (patrz punkt 5).

Obviously if there were no payments from the requestor, Concent should charge him with the full cost of all subtasks for which provider has submitted acceptances (minus the amount covered by forced payments made by Concent if any).

I think it would be best to move this discussion to a separate issue.

Reply:
#982. #983, #984